nwbwidgets/base.py

```diff
@@ -213,51 +212,52 @@ def plot_traces(time_series: TimeSeries, time_start=0, time_duration=None, trace
         Index range of traces to view
     title: str, optional
     ylabel: str, optional
-
     Returns
     -------
-
     """
-
+    
+    if type(time_series.data) != np.ndarray:
+        time_series.data = np.asarray(time_series.data)
+    
     if time_start == 0:
         t_ind_start = 0
     else:
         t_ind_start = timeseries_time_to_ind(time_series, time_start)
+    
     if time_duration is None:
         t_ind_stop = None
     else:
         t_ind_stop = timeseries_time_to_ind(time_series, time_start + time_duration)
-
-    if trace_window is None:
-        trace_window = [0, time_series.data.shape[1]]
-    tt = get_timeseries_tt(time_series, t_ind_start, t_ind_stop)
-    if time_series.data.shape[1] == len(tt):  # fix of orientation is incorrect
-        mini_data = time_series.data[trace_window[0]:trace_window[1], t_ind_start:t_ind_stop].T
+    
+    if t_ind_stop == None:
+        tt = get_timeseries_tt(time_series, t_ind_start, time_series.data.shape[-1])
     else:
-        mini_data = time_series.data[t_ind_start:t_ind_stop, trace_window[0]:trace_window[1]]
-
+        tt = get_timeseries_tt(time_series, t_ind_start, t_ind_stop)
+    
+    if trace_window is None:
+        trace_window = [0, time_series.data.shape[-1]]
+    
+    mini_data = time_series.data[:, t_ind_start:t_ind_stop]
+    
     gap = np.median(np.nanstd(mini_data, axis=0)) * 20
-    offsets = np.arange(trace_window[1] - trace_window[0]) * gap
-
+    offsets = np.arange(t_ind_stop - t_ind_start) * gap
     fig, ax = plt.subplots()
     ax.figure.set_size_inches(12, 6)
-    ax.plot(tt, mini_data + offsets)
+    ax.plot(tt, (mini_data + offsets).T)
```

Comment:
I think this was right before. TimeSeries.data objects are always time first: https://github.com/NeurodataWithoutBorders/nwb-schema/blob/5d6160eccc6e4c786fc328c8f1d227420f933bda/core/nwb.base.yaml#L69-L79

Reply:
Correct.